src/main/java/org/reactome/release/dataexport/FTPFileUploader.java

```diff
@@ -0,0 +1,468 @@
+	 * methods.
+	 *
+	 * @return Logger object for logging class activity
+	 */
+	protected Logger getLogger() {
+		return logger;
+	}
+
+	/**
+	 * Initialization for the FTP Server does the following:
+	 *
+	 * 1) Connects to the FTP Server
+	 * 2) Enters local passive mode
+	 * 3) Logs into the FTP Server with the username and password from getUserName() and getPassword() methods,
+	 * respectively
+	 * 4) Changes to the working directory obtained from the getReactomeDirectoryPathOnFTPServer() method
+	 *
+	 * @throws IOException Thrown if unable, for the FTP Server, to connect, log in, or change working directory
+	 * @see #getUserName()
+	 * @see #getPassword()
+	 * @see #getReactomeDirectoryPathOnFTPServer()
+	 */
+	public void initializeFTPConnectionToServer() throws IOException {
+		getFtpClientToServer().connect(getServerHostName());
+
+		getFtpClientToServer().enterLocalPassiveMode();
+
+		if (loginToFTPServer()) {
+			logger.info("Login successful to " + getServerHostName());
+		} else {
+			logger.error("Login to " + getServerHostName() + " failed");
+		}
+
+		getFtpClientToServer().changeWorkingDirectory(getReactomeDirectoryPathOnFTPServer());
+	}
+
+	/**
+	 * Updates (uploads new files and deletes old files) on the FTP Server and logs the files which exist on the FTP
+	 * Server in the Reactome specific directory after the update is complete.
+	 *
+	 * @throws IOException Thrown if unable to upload new Reactome files to the FTP server, delete old Reactome files
+	 * from the FTP server, provide the listing of files on the FTP Server after the update is complete
+	 */
+	public void updateFilesOnServer() throws IOException {
+		if (!uploadFilesToServer()) {
+			return;
+		}
+		deleteOldFilesFromServer();
+
+		logListingOfReactomeFilesPresentOnServer();
+		closeFTPConnectionToServer();
+	}
+
+	/**
+	 * Uploads the profile and links files (which match the pattern(s) defined by the method "isCurrentFile") for
+	 * Reactome data to the FTP Server.  Returns true if and only if all files are successfully uploaded; false
+	 * otherwise.
+	 *
+	 * @return <code>true</code> if all local files intended for the FTP Server are uploaded successfully,
+	 * <code>false</code> otherwise (including if some files are uploaded successfully, but at least one was not)
+	 * @throws IOException Thrown if unable to
+	 *  1) Get the names of the files from the local file system to upload
+	 *  2) Create an input stream to a local file to upload
+	 *  3) Store/upload the file on the FTP Server
+	 *  4) Get the current status from the FTP Server if attempting to upload a file fails without a thrown exception
+	 */
+	public boolean uploadFilesToServer() throws IOException {
+		List<String> filesToUpload = getLocalFileNamesToUpload();
+
+		if (filesToUpload.isEmpty()) {
+			getLogger().error(
+				"No files were found in the directory '" + getLocalOutputDirectoryPath() +
+					"' which should be uploaded to " + getServerHostName())
+			;
```

Comment:
Was this hard-to-read formatting _really_ necessary? Why is that `;` on its own line all by itself?

Reply:
I've redone the formatting using the `{}` placeholders.  The `;` on its own line was a mistake :)!

Reply:
Done: https://github.com/reactome/release-data-exporter/pull/4/commits/6e29c856511fee1cb5c5329b9d8706a9cc10fe69